benchto-driver/src/main/java/com/teradata/benchto/driver/execution/ExecutionSynchronizer.java

```diff
@@ -69,6 +107,40 @@ public void awaitAfterBenchmarkExecutionAndBeforeResultReport(Benchmark benchmar
         }
     }
 
+    /**
+     * Executes {@code callable} when time comes. The {@code callable} gets executed immediately, without
+     * offloading to a backghround thread, if execution time requested has already passed.
+     */
+    public <T> CompletableFuture<T> execute(Instant when, Callable<T> callable)
+    {
+        if (!Instant.now().isBefore(when)) {
```

Comment:
It would be, but it's not equivalent. I'd need `when.isBeforeOrEqual(now)` but there isn't any `isBeforeOrEqual` method)